sdk/include/sdk/grpc/GrpcDataPointValueProvider.h

```diff
@@ -27,7 +27,7 @@ class GrpcDataPointValueProvider : public IDataPointValueProvider {
 public:
     explicit GrpcDataPointValueProvider(sdv::databroker::v1::Datapoint datapoint);
 
-    DataPointFailure         getDataPointFailure() const override;
+    DataPointValue::Failure  getDataPointValueFailure() const override;
```

Comment:
Simply call this `getFailure`? It already is a Value Provider for a single datapoint, hence the name can be short and concise.